### PR TITLE
Task 7: Implement Formik library for add and edit movie

### PR DESCRIPTION
**Pull request by bandax**

### CODE REVIEW

File: src/components/AddMovieWindow/AddMovieWindow.tsx [hunk 1]
```diff
@@ -159,140 +178,149 @@ const AddMovieWindow: React.FunctionComponent<IAddMovieWindowProps> = (
...
+            };
+
+            props.onAddMovieSubmit(movieToSave);
+          }}
+        >
+          <Form>
+            <div className="add-movie-content">
+              <div className="row">
+                <div className="col-6">
+                  {showMovieIdField()}
```
Comment (endiny) [inline src/components/AddMovieWindow/AddMovieWindow.tsx:222]: Technically, user wouldn't compose id by himself. I'd suggest generating the id at service side

File: src/components/AddMovieWindow/AddMovieWindow.tsx [hunk 2]
```diff
@@ -15,51 +16,97 @@ interface IAddMovieWindowProps {
...
+      {meta.touched && meta.error ? (
+        <div className="error">{meta.error}</div>
+      ) : null}
+    </>
   );
-  const [releaseDate, setReleaseDate] = React.useState(
-    props.movie?.release_date ? new Date(props.movie.release_date) : new Date()
+};
+
+const FormikDatePicker = ({ ...props }) => {
```
Comment (endiny) [inline src/components/AddMovieWindow/AddMovieWindow.tsx:54]: Also, name it `MovieDatePicket` to be consistent with `MovieTextInput`

File: src/components/AddMovieWindow/AddMovieWindow.tsx [hunk 3]
```diff
@@ -15,51 +16,97 @@ interface IAddMovieWindowProps {
...
+        <div className="error">{meta.error}</div>
+      ) : null}
+    </>
   );
-  const [genres, setGenres] = React.useState(props.movie?.genres ?? []);
-  const [optionGenres, setOptionGenres] = React.useState([]);
-  const [selectedGenres, setSelectedGenres] = React.useState([]);
+};
+
+const FormikMultiSelect = ({ ...props }) => {
```
Comment (endiny) [inline src/components/AddMovieWindow/AddMovieWindow.tsx:75]: Please, rename it as well

File: src/components/AddMovieWindow/AddMovieWindow.tsx [hunk 4]
```diff
@@ -159,140 +178,149 @@ const AddMovieWindow: React.FunctionComponent<IAddMovieWindowProps> = (
...
+
+                  <MovieTextInput
+                    id="title"
+                    name="title"
+                    label="Title"
+                    placeholder="Title"
+                    type="text"
+                  />
+
+                  <label className="label-text" htmlFor="releaseDate">
```
Comment (endiny) [inline src/components/AddMovieWindow/AddMovieWindow.tsx:232]: I bet label can be placed into the form component as well

**Review by endiny — CHANGES_REQUESTED** [review]

File: src/components/AddMovieWindow/AddMovieWindow.tsx [hunk 5]
```diff
@@ -15,51 +16,97 @@ interface IAddMovieWindowProps {
...
+
+const AddMovieWindow: React.FunctionComponent<IAddMovieWindowProps> = (
+  props: IAddMovieWindowProps
+) => {
+  const [genreOptions, setOptionGenres] = React.useState([]);
 
   React.useEffect(() => {
-    const optGenres: IOptions[] = props.clasificationMovies.map(
+    const optGenres: IOption[] = props.clasificationMovies.map(
       (clasOption: IClasification) => ({
```
Comment (endiny) [inline src/components/AddMovieWindow/AddMovieWindow.tsx:110]: `clasOption`
typo?

File: src/components/AddMovieWindow/AddMovieWindow.tsx (hunk 1 above)
Comment (bandax) [reply src/components/AddMovieWindow/AddMovieWindow.tsx:222]: This function is to show/hide the movie id input checking if the movie is new to add or edit an existing one

File: src/components/AddMovieWindow/AddMovieWindow.tsx (hunk 5 above)
Comment (bandax) [reply src/components/AddMovieWindow/AddMovieWindow.tsx:110]: this is an alias for my map to get each element from the catalog of classification movies

File: src/components/AddMovieWindow/AddMovieWindow.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (bandax) [reply src/components/AddMovieWindow/AddMovieWindow.tsx]: Created "Shared" folder where I put all the formik controls

File: src/components/AddMovieWindow/AddMovieWindow.tsx [hunk 6]
```diff
@@ -159,140 +178,149 @@ const AddMovieWindow: React.FunctionComponent<IAddMovieWindowProps> = (
...
+            runtime: Yup.number().required('Required').positive().integer(),
+            budget: Yup.number().required('Required').positive().integer(),
+            revenue: Yup.number().required('Required').positive().integer(),
+            voteAverage: Yup.number().required('Required').positive().integer(),
+            voteCount: Yup.number().required('Required').positive().integer(),
+            overview: Yup.string().required('Required'),
+            tagline: Yup.string().required('Required'),
+            url: Yup.string().required('Required').url(),
+            releaseDate: Yup.date().default(() => new Date()),
+          })}
```
[the remark this answers is not in the document]
Comment (bandax) [reply src/components/AddMovieWindow/AddMovieWindow.tsx]: done

File: src/components/AddMovieWindow/AddMovieWindow.tsx [hunk 7]
```diff
@@ -159,140 +178,149 @@ const AddMovieWindow: React.FunctionComponent<IAddMovieWindowProps> = (
...
+              id: values.movieId,
+              overview: values.overview,
+              poster_path: values.url,
+              release_date: values.releaseDate.toISOString(),
+              revenue: values.revenue,
+              runtime: values.runtime,
+              tagline: values.tagline,
+              title: values.title,
+              vote_average: values.voteAverage,
+              vote_count: values.voteCount,
```
[the remark this answers is not in the document]
Comment (bandax) [reply src/components/AddMovieWindow/AddMovieWindow.tsx]: done

File: src/components/AddMovieWindow/AddMovieWindow.tsx (hunk 2 above)
Comment (bandax) [reply src/components/AddMovieWindow/AddMovieWindow.tsx]: rename using Formik- prefix

File: src/components/AddMovieWindow/AddMovieWindow.tsx (hunk 3 above)
Comment (bandax) [reply src/components/AddMovieWindow/AddMovieWindow.tsx]: rename using Formik- prefix

File: src/components/AddMovieWindow/AddMovieWindow.tsx (hunk 4 above)
Comment (bandax) [reply src/components/AddMovieWindow/AddMovieWindow.tsx]: Moved to the component

File: src/components/AddMovieWindow/AddMovieWindow.tsx (hunk 5 above)
Comment (endiny) [reply src/components/AddMovieWindow/AddMovieWindow.tsx]: I see
optional: it's cla_ss_ification

File: src/components/AddMovieWindow/AddMovieWindow.tsx (hunk 1 above)
Comment (endiny) [reply src/components/AddMovieWindow/AddMovieWindow.tsx:147]: But as far as I got it shows the input. If id is not supposed to be edited, please either make input disabled or show id in some other element like span

File: src/components/AddMovieWindow/AddMovieWindow.tsx [hunk 8]
```diff
@@ -159,140 +123,128 @@ const AddMovieWindow: React.FunctionComponent<IAddMovieWindowProps> = (
...
-                className="input-text"
-                id="overview"
-                name="overview"
-                placeholder="Overview here"
-                value={values.overview}
-                onChange={handleChange}
-              />
+        <Formik
+          initialValues={initialValues}
+          validationSchema={Yup.object(validations)}
```
Comment (endiny) [inline src/components/AddMovieWindow/AddMovieWindow.tsx:128]: Move `Yup.object` to `validations` declaration as well

**Pull request merged**